src/renderers/dom/client/eventPlugins/BeforeInputEventPlugin.js

```diff
@@ -457,20 +457,21 @@ var BeforeInputEventPlugin = {
     nativeEvent,
     nativeEventTarget
   ) {
-    return [
-      extractCompositionEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-      extractBeforeInputEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-    ];
+    var compose = extractCompositionEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    var before = extractBeforeInputEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    return compose && before ? [compose, before] : compose ? compose : before;
```

Comment:
Actually, if I'm interpreting the coverage correctly, BeforeInputEventPlugin has pretty weak coverage:

<img width="1439" alt="screen shot 2016-09-10 at 2 36 42 pm" src="https://cloud.githubusercontent.com/assets/590904/18412718/017c8ebc-7764-11e6-8228-7ac1b9489129.png">

<img width="648" alt="screen shot 2016-09-10 at 2 35 50 pm" src="https://cloud.githubusercontent.com/assets/590904/18412721/03e3fa5a-7764-11e6-8f7b-df55acaa0bdd.png">

It's possible this is just really hard to test (keyboard stuff). Looks like neither event types (composition or beforeInput) ever instantiate.